test/Snapshot/InMemorySnapshotRepositoryTest.php

```diff
@@ -13,6 +13,31 @@
 class InMemorySnapshotRepositoryTest extends SnapshotRepositoryTest
 {
     /**
+     * @test
+     */
+    public function it_does_not_mutate_state_of_Snapshot_Aggregate()
+    {
+        $aggregate = $this->createAggregateWithHistory(5);
+        $this->repository->save(new Snapshot($aggregate));
+
+        // Applying another event to Snapshotted Aggregate should not affect Snapshot version
+        $aggregate->apply(new MyEvent());
+        $aggregate->getUncommittedEvents();
+
+        $snapshot = $this->repository->load(42);
+        $this->assertEquals(new Snapshot($this->createAggregateWithHistory(5)), $snapshot);
+
+        $loadedAggregate = $snapshot->getAggregateRoot();
```

Comment:
I had them there initially, but I figured this case would only be relevant for InMemory implementations, hence the split. 

Couldn't see how it would be a problem when you're using mongodb or redis for example. There shouldn't be any references to the original object to begin with

If you want, I could put move them back into the parent class just for convenience's sake.